source/ui/data/VerticalDelegate.js

```diff
@@ -401,13 +401,28 @@
 			// the only time we don't refresh is if the first index of the contiguous set of added
 			// models is beyond our final rendered page (possible) indices
 
-			// in the case where it does not need to refresh the existing controls it will update its
-			// measurements and page positions within the buffer so scrolling can continue properly
+			// in the case where it does not need to refresh the existing controls except the last page
+			// if the last page is not fully filled, it will be filled with added models
+			// so we should generate the last page.
+
+			// it will update its measurements and page positions within the buffer
+			// so scrolling can continue properly
 
 			// if we need to refresh, do it now and ensure that we're properly setup to scroll
 			// if we were adding to a partially filled page
 			if (props.index <= end ) this.refresh(list);
 			else {
+				// we should confirm that the page which new models are added is need to update list.metrics
+				var targetPageIndex = this.pageForIndex(list, props.index),
+					page = list.metrics.pages[targetPageIndex],
+					sp = list.psizeProp,
+					pc = this.pageCount(list),
+					pageSize = this.defaultPageSize(list);
+
+				if (targetPageIndex < pc && page[sp] < pageSize) {
```

Comment:
According to the bug description, wouldn't this only need to happen if the page was the last page? If it wasn't the last page, then we know it is already full. In that case, we wouldn't need to update this value or call `this.pageCount` or `this.defaultPageSize` needlessly. We also wouldn't need to check `page[sp] < pageSize` and could safely just assign the value (would fix the issue in problem case and be a nop in regular case).